### PR TITLE
platform guide updates

### PR DESCRIPTION
**Pull request by rohitthakur2590**
Signed-off-by: rohitthakur2590 <rohitthakur2590@outlook.com>

##### SUMMARY

Add platform guide for VyOS.

##### ISSUE TYPE

 
- Feature Pull Request

##### COMPONENT NAME

platform_vyos.rst
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: docs/docsite/rst/network/user_guide/platform_vyos.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+VyOS supports Enable Mode (Privilege Escalation). This page offers details on how to use Enable Mode on VyOS in Ansible.
+
+.. contents:: Topics
+
+Connections Available
+================================================================================
+
+.. table::
+    :widths: auto
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_vyos.rst]:
```suggestion
```

Comment (samccann) [reply docs/docsite/rst/network/user_guide/platform_vyos.rst]: a recent PR removed using this :widths: setting

**Review by samccann — COMMENTED** [review]
You also need to add the platform_vyos file to https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/network/user_guide/platform_index.rst

**Comment by samccann** [comment]
add the new platform file to platform_index.rst (in alphabetical order) to fix that not-in-toc tree warning.

**Review by danielmellado — APPROVED** [review]
LGTM but I'd like to see it rendered. Do we have any gate on shippable to actually check this out?

**Comment by danielmellado** [comment]
Also I think we should add some triggers to irrelevant-files in order to avoid the Zuul tests to be run in documentation cases.

File: docs/docsite/rst/network/user_guide/platform_vyos.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                          accepts ``-u myuser -k`` if using password
+
+    Indirect Access       via a bastion (jump host)
+
+    Connection Settings   ``ansible_connection: network_cli``
+
+    |enable_mode|         supported: use ``ansible_become: yes`` with
+                          ``ansible_become_method: enable`` and ``ansible_become_password:``
+
+    Returned Data Format  ``stdout[0].``
```
Comment (NilashishC) [inline docs/docsite/rst/network/user_guide/platform_vyos.rst]: Returned data format is not same for all the modules. `Refer to individual module documentation` sounds better to me.

File: docs/docsite/rst/network/user_guide/platform_vyos.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    Credentials           uses SSH keys / SSH-agent if present
+
+                          accepts ``-u myuser -k`` if using password
+
+    Indirect Access       via a bastion (jump host)
+
+    Connection Settings   ``ansible_connection: network_cli``
+
+    |enable_mode|         supported: use ``ansible_become: yes`` with
```
Comment (NilashishC) [inline docs/docsite/rst/network/user_guide/platform_vyos.rst]: VyOS doesn't support `enable` mode.

**Review by NilashishC — CHANGES_REQUESTED** [review]

**Comment by danielmellado** [comment]
https://github.com/ansible/ansible-zuul-jobs/pull/186 Added such change ;)

**Review by ganeshrn — APPROVED** [review]
lgtm. Thank you!

**Review by NilashishC — APPROVED** [review]

**Pull request merged**